cli/azd/pkg/environment/environment.go

```diff
@@ -36,14 +37,78 @@ const ContainerRegistryEndpointEnvVarName = "AZURE_CONTAINER_REGISTRY_ENDPOINT"
 const ResourceGroupEnvVarName = "AZURE_RESOURCE_GROUP"
 
 type Environment struct {
-	// Values is a map of setting names to values.
-	Values map[string]string
+	// values is a map of setting names to values.
+	// all keys are converted to upper case before it is set.
+	values map[string]string
 	// File is a path to the file that backs this environment. If empty, the Environment
 	// will not be persisted when `Save` is called. This allows the zero value to be used
 	// for testing.
 	File string
 }
 
+// DeleteVariable removes an entry from azd environment which key matches the variable name.
+func (e *Environment) DeleteVariable(variableName string) {
+	delete(e.values, strings.ToUpper(variableName))
+}
+
+// SetVariable update or create an entry to the azd environment.
+// key is converted to upper case.
+func (e *Environment) SetVariable(key, value string) {
+	e.values[strings.ToUpper(key)] = value
+}
+
+// GetValue get the value of an entry with the requested key.
+// return empty string and false if the key is not found.
+// Use ValueOf if you don't need to check if value is found
+func (e *Environment) GetValue(key string) (string, bool) {
```

Comment:
For what it's worth, if we want to draw inspiration from `os` handling of environment variables, it would be `Lookup` for exist vs `Get`